### ISSUE TITLE
Setting a `Parameter` value in `__init__` before calling `super().__init__`

### ISSUE THREAD

**Issue by maximlt** [holoviz/param#815]
Working on the private namespace I have had to handle when a `Parameter` value is set in `__init__` before calling `super().__init__`. This is to me an anti-pattern and I would be very happy not having to support that in Param. Also, it sort of works but isn't really fully tested so I wouldn't be surprised if it was also partially broken.

The only occurrence of this patter I found across the HoloViz packages was in HoloViews and it's been updated since then in https://github.com/holoviz/holoviews/pull/5814.

I'd be in favor of deprecating this usage.

**Comment by jbednar** [comment]
I agree with deprecating this usage; it is not something I would expect to be fully supported and it is difficult to imagine a valid need for it.
